# Задание 0

Музыкальный фрагмент длительностью 8 секунд был записан в формате моно, частотой дискретизации 10000 Гц и 16-ти битным разрешением. Сжатие данных не производилось. Определите объем получившегося файла в КБайтах. В ответе укажите только целую часть.

<h3><details>
	<summary>Ответ</summary>
    156
</details></h3>

In [2]:
8 * 1 * 10000 * 16 / (2 ** 10 * 8)

156.25

In [ ]:
моно - 1
стерео - 2
квадро - 4

# Задание 1

Производится звукозапись музыкального фрагмента в формате стерео (двухканальная запись) с частотой дискретизации 32 кГц и 32-⁠битным разрешением. Результаты записываются в файл, сжатие данных не производится; размер полученного файла 40 Мбайт. Затем производится повторная запись этого же фрагмента в формате моно (одноканальная запись) с частотой дискретизации 16 кГц и 16-⁠битным разрешением. Сжатие данных не производилось.

Укажите размер файла в Мбайт, полученного при повторной записи. В ответе запишите только целое число, единицу измерения писать не нужно.

<h3><details>
	<summary>Ответ</summary>
	Получаем систему:

40 (Мбайт)  =  2 (количество каналов) · 32000 (частота) · 4 (разрешение в байтах) · X (время в секундах).

V (Мбайт)  =  1 (количество каналов) · 16000 (частота) · 2 (разрешение в байтах) · X (время в секундах).

 

Тогда v  =  40 : (2 · 2 · 2)  =  5 Мбайт.
</details></h3>

In [ ]:
2 * 32000 * 32 * t = 40 * 2 ** 23
1 * 16000 * 16 * t = v
2   2       2      = v * 2 * 2 * 2 = 40 * 2 ** 23

# Задание 2

Для проведения эксперимента записывается звуковой фрагмент в формате квадро (четырёхканальная запись) с частотой дискретизации 32 кГц и 32-⁠битным разрешением. Результаты записываются в файл, сжатие данных не производится; дополнительно в файл записывается служебная информация, необходимая для эксперимента, размер полученного файла 97 Мбайт.

Затем производится повторная запись этого же фрагмента в формате моно (одноканальная запись) с частотой дискретизации 16 кГц и 16-⁠битным разрешением. Результаты тоже записываются в файл без сжатия и со служебной информацией, размер полученного файла 7 Мбайт. Объём служебной информации в обоих случаях одинаков. Укажите этот объём в мегабайтах.

В ответе укажите только число (количество Мбайт), единицу измерения указывать не надо.

<h3><details>
	<summary>Ответ</summary>
Получаем систему:

<p class="left_margin">97 · 2<sup>20</sup> (байт)  =  4 (ко&shy;ли&shy;че&shy;ство ка&shy;на&shy;лов) · 32000 (ча&shy;сто&shy;та) · 4 (раз&shy;ре&shy;ше&shy;ние в бай&shy;тах) · <i>T</i> (время в се&shy;кун&shy;дах) + <i>X</i> (объём слу&shy;жеб&shy;ной ин&shy;фор&shy;ма&shy;ции);</p>

<p class="left_margin">7 · 2<sup>20</sup> (байт)  =  1 (ко&shy;ли&shy;че&shy;ство ка&shy;на&shy;лов) · 16000 (ча&shy;сто&shy;та) · 2 (раз&shy;ре&shy;ше&shy;ние в бай&shy;тах) · <i>T</i> (время в се&shy;кун&shy;дах) + <i>X</i> (объём слу&shy;жеб&shy;ной ин&shy;фор&shy;ма&shy;ции).</p>
</details></h3>

In [ ]:
97 = 4 * 2 * 2 * t + x
7 = t + x

# Задание 3

Музыкальный фрагмент был записан в формате стерео (двухканальная запись), оцифрован и сохранён в виде файла без использования сжатия данных. Размер полученного файла без учёта размера заголовка файла  — 48 Мбайт. Затем тот же музыкальный фрагмент был записан повторно в формате моно и оцифрован с разрешением в 1,5 раза выше и частотой дискретизации в 3 раза меньше, чем в первый раз. Сжатие данных не производилось. Укажите размер в Мбайт файла, полученного при повторной записи. В ответе запишите только целое число, единицу измерения писать не нужно. Искомый объём не учитывает размера заголовка файла.

<h3><details>
	<summary>Ответ</summary>
    48 * 1.5 / (3 * 2) = 12
</details></h3>

# Задание 4

Голосовое сообщение продолжительностью 90 с было записано в формате стерео и оцифровано с глубиной кодирования 16 бит и частотой дискретизации 48 000 измерений в секунду. Сжатие данных не использовалось. Файл с оцифрованным голосовым сообщением был передан по каналу связи, пропускная способность которого 3200 бит/с. Сколько секунд длилась передача файла? В ответе запишите целое число, единицу измерения указывать не нужно.

<h3><details>
	<summary>Ответ</summary>
    2 * 16 * 48000 * 90 / 3200 = 43200
</details></h3>

# Задание 5

Маша скачивает из Интернета альбом любимой группы, оцифрованный в формате стерео с частотой дискретизации 48 000 Гц и разрешением 34 бит без использования сжатия. В альбоме 13 треков общей длительностью 42 минуты 20 секунд. Каждый трек содержит заголовок объемом 110 Кбайт. Сколько секунд будет скачиваться альбом по каналу связи со скоростью передачи данных 314 572 800 бит⁠/⁠с? В ответе запишите целую часть полученного числа.

<h3><details>
	<summary>Ответ</summary>
    Посчитаем общий объем альбома:

2 * 48000 * 34 * 2540 = 8_290_560_000

Посчитаем общий объем заголовков:

13 * 100 * 1024 * 8 = 11_714_560

Посчитаем время передачи:

(8_290_560_000 + 11_714_560) / 314_572_800 = 26.39

26

</details></h3>

# Задание 6

Прибор автоматической фиксации нарушений правил дорожного движения делает цветные фотографии размером 1024×768 пикселей. Снимки сохраняются в памяти камеры, группируются в пакеты по 27 штук, затем передаются в центр обработки информации со скоростью передачи данных 1 638 400 бит/с. Каково максимально возможное количество цветов в палитре изображения, если на передачу одного пакета отводится не более 220 секунд?

In [5]:
# x * 1024 * 768 * 27 / 1_638_400 = 200

x = 220 * 1_638_400 / (1024 * 768 * 27)

2 ** int(x)

65536

<h3><details>
	<summary>Ответ</summary>
    65536
</details></h3>

# Задание 7

Прибор автоматической фиксации нарушений правил дорожного движения делает цветные фотографии размером 1024×768 пикселей, используя палитру из 4096 цветов. Снимки сохраняются в памяти камеры, группируются в пакеты по несколько штук, а затем передаются в центр обработки информации со скоростью передачи данных 1 310 720 бит/с. Каково максимально возможное количество снимков в одном пакете, если на передачу одного пакета отводится не более 300 секунд? В ответе запишите целое число.

<h3><details>
	<summary>Ответ</summary>
    41
</details></h3>

# Задание 8

Для сканирования документов с мелким текстом используют сканер с разрешением 600 dpi (точек на дюйм). Документы — листы А4 размером 8.27×11.69 дюймов. Сканер создаёт изображение, каждый пиксель которого может иметь один из 256 оттенков серого цвета.

Какое минимальное количество Мбайт нужно выделить для сохранения 10 документов? Документ сканируется только с одной стороны.

В ответе укажите только целое число.

<h3><details>
	<summary>Ответ</summary>
    332
</details></h3>

# Задание 9

<div class="Markdown-module__Markdown--f5ALc TasksBankTaskCard-module__TaskCardContentMarkdown--hsXqP"><p>Изображения размером 4×7 дюйма сканируются с&nbsp;разрешением 600&nbsp;ppi и&nbsp;использованием 2¹⁶&nbsp;цветов.</p>
<p>Сколько таких изображений можно полностью передать с&nbsp;помощью модема, скорость которого равна 1000000&nbsp;бит/с, за&nbsp;10&nbsp;минут? В&nbsp;ответе запишите только целое число.</p></div>

<h3><details>
	<summary>Ответ</summary>
    3
</details></h3>

# Задание 10

<p>При кодировании растрового изображения для каждого пикселя используется палитра из&nbsp; <span class="math math-inline"><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msup><mn>2</mn><mn>24</mn></msup></mrow><annotation encoding="application/x-tex">2^{24}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height: 0.8141em;"></span><span class="mord"><span class="mord">2</span><span class="msupsub"><span class="vlist-t"><span class="vlist-r"><span class="vlist" style="height: 0.8141em;"><span style="top: -3.063em; margin-right: 0.05em;"><span class="pstrut" style="height: 2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">24</span></span></span></span></span></span></span></span></span></span></span></span></span> &nbsp;цветов и&nbsp;256&nbsp;уровней прозрачности. Коды пикселей записываются в&nbsp;файл один за&nbsp;другим без промежутков. Какой минимальный объём памяти (в&nbsp;Кбайт) нужно зарезервировать, чтобы можно было сохранить любое растровое изображение размером 1024&nbsp;на&nbsp;768 пикселей?</p>

In [7]:
(24 + 8) * 1024 * 768 / 8 / 1024

3072.0

<h3><details>
	<summary>Ответ</summary>
    3072
</details></h3>

# Задание 11

Вова делает логотипы для сайтов размером 640 на 128 пикселей и хранит их в папке. Для неё он выделил на жёстком диске 10 Мбайт памяти без учёта размера заголовка файла. Каждый файл хранится в двух форматах — превью и оригинал. Файл оригинального изображения больше превью на 65%. В палитре используется 1024 цвета. Для кодирования цвета каждого пикселя используется одинаковое количество бит, коды пикселей записываются в файл один за другим без промежутков.

Какое максимальное количество логотипов может сохранить Вова?

In [17]:
10 / (640 * 128 * 10 * (1 + 1 / 1.65) / 8 / 1024 / 1024)

63.75849056603774

<h3><details>
	<summary>Ответ</summary>
    63
</details></h3>

# Задание 12

Для хранения сжатого произвольного растрового изображения размером 320 на 512 пикселей отведено 50 Кбайт памяти без учёта размера заголовка файла. Файл оригинального изображения больше сжатого на 55%. Для кодирования цвета каждого пикселя используется одинаковое количество бит, коды пикселей записываются в файл один за другим без промежутков.

Какое максимальное количество цветов можно использовать в изображении?

In [9]:
50 * 1024 * 8 * 1.55 / (320 * 512)

3.875

<h3><details>
	<summary>Ответ</summary>
    8
</details></h3>

# Задание 13

<p>При кодировании произвольного сжатого растрового изображения размером 1280×1024 пикселей используется палитра из&nbsp;2³²&nbsp;цветов. Коды пикселей записываются в&nbsp;файл один за&nbsp;другим без промежутков. После сохранения информации о&nbsp;пикселях изображение сжимается. Размер итогового файла после сжатия на&nbsp;40%&nbsp;меньше исходного. Определите, какой минимальный объем памяти в&nbsp;Мегабайтах необходимо выделить для хранения такого изображения.</p>

In [11]:
1280 * 1024 * 32 * 0.6 / 8 / 1024 / 1024

3.0

<h3><details>
	<summary>Ответ</summary>
    3
</details></h3>

# Задание 14

Для хранения сжатого произвольного растрового изображения размером 640 на 860 пикселей отведено 500 Кбайт памяти без учёта размера заголовка файла. Файл оригинального изображения больше сжатого на 23%. Для кодирования цвета каждого пикселя используется одинаковое количество бит, коды пикселей записываются в файл один за другим без промежутков. Какое максимальное количество цветов можно использовать в изображении?

In [18]:
# 640 * 860 * x = 500 * 8 * 1024 * 1.23

500 * 8 * 1024 * 1.23 / (640 * 860)


9.153488372093022

<h3><details>
	<summary>Ответ</summary>
    512
</details></h3>

# Задание 15

Для хранения сжатого растрового изображения выделено 3 Мбайт. Для каждого пикселя записывается информация о его цвете и уровне прозрачности. Как информация о цвете, так и информация об уровне прозрачности записываются с помощью минимального и одинакового для каждой части бит. После кодирования информации о каждом пикселе изображение сжимается. Сжатое изображение меньше исходного на 20%. Определите, какое максимальное количество уровней прозрачности может быть у изображения размером 1080×920, если известно, что используется цветовая палитра, содержащая 1 миллион цветов.

In [19]:
2 ** 20

1048576

In [21]:
# (20 + x) * 1080 * 920 = 3 * 8 * 1024 * 1024 / 0.8

x = 3 * 8 * 1024 * 1024 / 0.8 / (1080 * 920) - 20

x

11.659903381642511

In [22]:
2 ** int(x)

2048

<h3><details>
	<summary>Ответ</summary>
    2048
</details></h3>